ui-tests/src/test/java/io/jenkins/plugins/analysis/warnings/WarningsPluginUiTest.java

```diff
@@ -0,0 +1,611 @@
+    public void should_record_issues_in_pipeline_and_expand_tokens() {
+        WorkflowJob job = jenkins.jobs.create(WorkflowJob.class);
+        job.sandbox.check();
+
+        createRecordIssuesStep(job, 1);
+
+        job.save();
+
+        Build referenceBuild = buildJob(job);
+
+        assertThat(referenceBuild.getConsole())
+                .contains("[total=4]")
+                .contains("[new=0]")
+                .contains("[fixed=0]")
+                .contains("[checkstyle=1]")
+                .contains("[pmd=3]");
+
+        job.configure(() -> createRecordIssuesStep(job, 2));
+
+        Build build = buildJob(job);
+
+        assertThat(build.getConsole())
+                .contains("[total=25]")
+                .contains("[new=23]")
+                .contains("[fixed=2]")
+                .contains("[checkstyle=3]")
+                .contains("[pmd=2]");
+
+        verifyPmd(build);
+        verifyFindBugs(build);
+        verifyCheckStyle(build);
+        verifyCpd(build);
+    }
+
+    private void createRecordIssuesStep(final WorkflowJob job, final int build) {
+        job.script.set("node {\n"
+                + createReportFilesStep(job, build)
+                + "recordIssues tool: checkStyle(pattern: '**/checkstyle*')\n"
+                + "recordIssues tool: pmdParser(pattern: '**/pmd*')\n"
+                + "recordIssues tools: [cpd(pattern: '**/cpd*', highThreshold:8, normalThreshold:3), findBugs()], aggregatingResults: 'false' \n"
+                + "def total = tm('${ANALYSIS_ISSUES_COUNT}')\n"
+                + "echo '[total=' + total + ']' \n"
+                + "def checkstyle = tm('${ANALYSIS_ISSUES_COUNT, tool=\"checkstyle\"}')\n"
+                + "echo '[checkstyle=' + checkstyle + ']' \n"
+                + "def pmd = tm('${ANALYSIS_ISSUES_COUNT, tool=\"pmd\"}')\n"
+                + "echo '[pmd=' + pmd + ']' \n"
+                + "def newSize = tm('${ANALYSIS_ISSUES_COUNT, type=\"NEW\"}')\n"
+                + "echo '[new=' + newSize + ']' \n"
+                + "def fixedSize = tm('${ANALYSIS_ISSUES_COUNT, type=\"FIXED\"}')\n"
+                + "echo '[fixed=' + fixedSize + ']' \n"
+                + "}");
+    }
+
+    private StringBuilder createReportFilesStep(final WorkflowJob job, final int build) {
+        String[] fileNames = {"checkstyle-result.xml", "pmd.xml", "findbugsXml.xml", "cpd.xml", "Main.java"};
+        StringBuilder resourceCopySteps = new StringBuilder();
+        for (String fileName : fileNames) {
+            resourceCopySteps.append(job.copyResourceStep(WARNINGS_PLUGIN_PREFIX
+                    + "build_status_test/build_0" + build + "/" + fileName).replace("\\", "\\\\"));
+        }
+        return resourceCopySteps;
+    }
+
+    /**
+     * Runs a freestyle job with all tools two times. Verifies the analysis results in several views.
+     */
+    @Test
+    public void should_show_build_summary_and_link_to_details() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'should_show_build_summary_and_link_to_details' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5151082)